File: solutions/cpp/sum-of-multiples/1/sum_of_multiples.cpp
#include "sum_of_multiples.h"

namespace sum_of_multiples
{

    uint32_t to(const std::set<uint32_t> &multiples_of, uint32_t exclusive_upper_bound)
    {
        std::set<uint32_t> multiples{};

        for (uint32_t base : multiples_of)
        {
            for (uint32_t multiple = base; multiple < exclusive_upper_bound; multiple += base)
            {
                multiples.insert(multiple);
            }
        }

        uint32_t sum = 0;
        for (uint32_t multiple : multiples)
        {
            sum += multiple;
        }

        return sum;
    }

} // namespace sum_of_multiples

File: solutions/php/sum-of-multiples/1/SumOfMultiples.php
<?php

declare(strict_types=1);

function sumOfMultiples(int $number, array $multiples): int
{
    if ($number < 2) return 0;
    $multiplesInRange = array_filter(range(1, $number - 1), function($v, $k) use ($multiples) {
        $isMultiple = false;
        foreach($multiples as $multiple){
            if ($multiple > 0 && $v % $multiple == 0) {
                $isMultiple = true;
                break;
            }
        }
        return $isMultiple;
    }, ARRAY_FILTER_USE_BOTH);

    return array_sum($multiplesInRange);
}
